cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/Reducer.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.api;
+
+import io.cdap.cdap.api.annotation.Beta;
+
+/**
+ * This class can be implemented by an {@link Aggregator} to reduce the number of group values in each split before
+ * grouping all the values for the group key to achieve better performance.
```

Comment:
Can you add some more detail, perhaps including an example? Plugin developers should know that this is different from a MapReduce 'Reducer', and that tis reduce method gets called both before and after the grouping. It should be clear that this will make the plugin perform better.